src/Microsoft.Health.Fhir.SqlServer.UnitTests/Features/Search/SqlQueryGeneratorTests.cs

```diff
@@ -0,0 +1,122 @@
+        SearchOptions searchOptions = new()
+        {
+            Sort = [],
+            ResourceVersionTypes = ResourceVersionType.Latest,
+        };
+
+        var output = _queryGenerator.VisitSqlRoot(sqlExpression, searchOptions);
+
+        Assert.Contains("IsHistory = 0", _strBuilder.ToString());
+        Assert.Contains("IsDeleted = 0", _strBuilder.ToString());
+    }
+
+    [Fact]
+    public void GivenASearchTypeForSoftDeletedOnly_WhenSqlGenerated_ThenFilterForSoftDeletedInSql()
+    {
+        Expression predicate = Expression.And([new SearchParameterExpression(new SearchParameterInfo("_type", "_type"), new StringExpression(StringOperator.Equals, FieldName.String, null, "Patient", false))]);
+        SqlRootExpression sqlExpression = new([new(null, predicate, SearchParamTableExpressionKind.All)], new List<SearchParameterExpressionBase>());
+        SearchOptions searchOptions = new()
+        {
+            Sort = [],
+            ResourceVersionTypes = ResourceVersionType.SoftDeleted,
+        };
+
+        var output = _queryGenerator.VisitSqlRoot(sqlExpression, searchOptions);
+
+        Assert.Contains("IsDeleted = 1", _strBuilder.ToString());
+    }
+
+    [Fact]
+    public void GivenASearchTypeForHistoryOnly_WhenSqlGenerated_ThenFilterForHistoryInSql()
+    {
+        Expression predicate = Expression.And([new SearchParameterExpression(new SearchParameterInfo("_type", "_type"), new StringExpression(StringOperator.Equals, FieldName.String, null, "Patient", false))]);
+        SqlRootExpression sqlExpression = new([new(null, predicate, SearchParamTableExpressionKind.All)], new List<SearchParameterExpressionBase>());
+        SearchOptions searchOptions = new()
+        {
+            Sort = [],
+            ResourceVersionTypes = ResourceVersionType.History,
+        };
+
+        var output = _queryGenerator.VisitSqlRoot(sqlExpression, searchOptions);
+
+        Assert.Contains("History = 1", _strBuilder.ToString());
+    }
+
+    [Fact]
+    public void GivenASearchTypeForLatestHistorySoftDeleted_WhenSqlGenerated_ThenFiltersArentInSql()
+    {
+        Expression predicate = Expression.And([new SearchParameterExpression(new SearchParameterInfo("_type", "_type"), new StringExpression(StringOperator.Equals, FieldName.String, null, "Patient", false))]);
+        SqlRootExpression sqlExpression = new([new(null, predicate, SearchParamTableExpressionKind.All)], new List<SearchParameterExpressionBase>());
+        SearchOptions searchOptions = new()
+        {
+            Sort = [],
+            ResourceVersionTypes = ResourceVersionType.Latest | ResourceVersionType.History | ResourceVersionType.SoftDeleted,
+        };
+
+        var output = _queryGenerator.VisitSqlRoot(sqlExpression, searchOptions);
+
+        Assert.DoesNotContain("IsHistory =", _strBuilder.ToString());
+        Assert.DoesNotContain("IsDeleted =", _strBuilder.ToString());
+    }
```

Comment:
Should there be one for the case of history and soft deleted are included but latest isn't?

Reply:
Can add that yeah 🧷

Reply:
Added